src/vue-property-decorator.ts

```diff
@@ -13,10 +14,27 @@ export type Constructor = {
  * @param  options the option for the prop
  * @return PropertyDecorator
  */
-export function Prop(options: (PropOptions | Constructor | Constructor[]) = { type: null }): PropertyDecorator {
-  return createDecorator((componentOptions, key) => {
-      (componentOptions.props || (componentOptions.props = {}) as any)[key] = options
-    })
+export function Prop(target: Vue, key: string): void
+export function Prop(target: (PropOptions | Constructor[])): PropertyDecorator
+export function Prop(options: (Vue | PropOptions | Constructor[]), key?: string): void | PropertyDecorator {
```

Comment:
You want to add the single `Constructor` so people can write `@Prop(Number) propA: number`, right? Well if you look in the tests that is already possible. It works in a weird way at the moment, because I think TypeScript just interprets a single `Constructor` input as a `PropOptions`... Hmm yeah that's probably error prone. I'll fix that. And yes I'll make empty parentheses work again.